### PR TITLE
IOS-2255: Implemented searchMonthlyFacilityParking GET Request

### PR DESCRIPTION
**Pull request by bdrelling**
**Issue Link**
IOS-2255

**Description**
- Added `SearchGetMonthlyFacilityRequest` following the paradigm from previous POCs.
- Added `Parameters` struct within this request for strongly typed parameter enforcement. It adheres to the `asParameterDictionary` function for making network requests simple.
- Added mock file for the request.
- Updated access levels for `searchMonthlyParking` while doing it for this PR.

Similar to the previous PR, no real testing of the actual search response included just yet, this is only meant to test the deserialization. Testing the full responses will likely involve mirroring the `SpotHeroAPI` v1 tests in the future.

### CODE REVIEW

File: Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright © 2020 SpotHero, Inc. All rights reserved.
+
+import Foundation
+import UtilityBeltNetworking
+
+/// Represents a request for fetching monthly facility.
```
Comment (khaptonstall) [inline Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:6]:
```suggestion
/// Represents a request for fetching a monthly facility.
```

File: Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public extension SearchGetMonthlyFacilityRequest {
+    /// Represents the query parameters used for fetching monthly facility.
+    struct Parameters: Encodable {
+        private enum CodingKeys: String, CodingKey {
+            case startDate = "starts"
+        }
+        
+        /// Start date from which results will be generated. The supported format is YYYY-MM-DD.
+        /// If this parameter is not provided, results will be generated from the date at which the request was received.
+        private let startDate: Date?
```
Comment (khaptonstall) [inline Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:45]: Just thinking, if all the documentation resides on private properties and not on the public initializer, it's not as easy for clients to understand what these parameters are and how they should be used.

File: Tests/SpotHeroAPITests/TestCases/RequestTests/SearchGetMonthlyFacilityRequestTests.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+@testable import SpotHeroAPINext
+import XCTest
+
+private protocol SearchGetMonthlyFacilityRequestTests: APITestCase {
+    func testGetMonthlyFacilitySucceeds()
+}
+
+private extension SearchGetMonthlyFacilityRequestTests {
+    /// Attempts to fetch a list of monthly facility, expecting success
```
Comment (khaptonstall) [inline Tests/SpotHeroAPITests/TestCases/RequestTests/SearchGetMonthlyFacilityRequestTests.swift:11]:
```suggestion
    /// Attempts to fetch a monthly facility, expecting success
```

File: Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    init(client: NetworkClient) {
+        self.client = client
+    }
+    
+    @discardableResult
+    public func callAsFunction(withID facilityID: Int,
+                               parameters: Parameters? = nil,
+                               completion: @escaping RequestCompletion<ResponseModel>) -> URLSessionTask? {
+        return self.client.request(
+            route: "\(Self.route)/\(facilityID)",
```
Comment (khaptonstall) [inline Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:26]: What's the plan for supporting routes such as `user/{user-id}/reservations/{reservation-id}`?

File: Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift (hunk 2 above)
Comment (bdrelling) [reply Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:45]: Good thought. I'll make a ticket for this on the initializers.

Comment (bdrelling) [reply Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:45]: https://spothero.atlassian.net/browse/IOS-2293

File: Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift (hunk 4 above)
Comment (bdrelling) [reply Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:26]: Right now, nothing. Have been thinking about this for the past hour or so. For now, we'll do this, but I'll look into how to do path parameters in a cleaner way now that I have real examples.

Comment (bdrelling) [reply Sources/SpotHeroAPI/Search/Monthly/Requests/SearchGetMonthlyFacilityRequest.swift:26]: https://spothero.atlassian.net/browse/IOS-2294

**Pull request merged**